examples/nodejs/storage/user-program.ts

```diff
@@ -9,82 +9,25 @@ async function main() {
-        case StoreValueType.STRING:
-          console.log('The value is a string:', getResponse.tryGetValueString());
-          break;
-        case StoreValueType.INTEGER:
-          console.log('The value is an integer:', getResponse.tryGetValueInteger());
-          break;
-        case StoreValueType.FLOAT:
-          console.log('The value is a float:', getResponse.tryGetValueFloat());
-          break;
-        case StoreValueType.BOOLEAN:
-          console.log('The value is a boolean:', getResponse.tryGetValueBoolean());
-          break;
-        default:
-          console.error('Unknown type');
-      }
-      // An alternative to this switch we have derive subclases of StoreGet.Success:
-      // StoreStringValue, StoreIntegerValue, StoreFloatValue, etc.
-      //In that world the developer can use more instanceof checks:
-      // if (getResponse instanceof StoreGet.Success) {
-      //    if (getResponse instanceof StoreStringValue) {
-      //      console.log('The value is a string:', getResponse.value);
-      //    } else if (getResponse instanceof StoreIntegerValue) {
-      //      console.log('The value is an integer:', getResponse.value);
-      //    } else if (getResponse instanceof StoreFloatValue) {
-      //      // ...
-      //    } else {
-      //      console.error('Unknown type');
-      //    }
-      // }
-      //
-      // I feel this pattern makes sense in a language with first-class pattern matching, such as
-      // Rust, Scala, Kotlin, etc.
-      // But in JavaScript, I feel the switch is at the right level of complexity.
-    } else if (getResponse instanceof StoreGet.Error) {
-      console.error('Error:', getResponse.message);
-    } else {
-      console.error('Unknown response type');
-    }
-  }
-
-  // I know what I'm doing
-  const response = await client.get(storeName, 'string-key');
-  if (response instanceof StoreGet.Success) {
-    console.log(`The value is: ${response.tryGetValueString()} which is the same as ${response.value as string}`);
-    console.log("If I don't know what I'm doing, I can catch the exception");
-    try {
-      console.log(response.tryGetValueInteger());
-    } catch (error: unknown) {
-      console.error(error);
+    switch (getResponse.response) {
+      case StoreGet.ResponseType.Success:
+        switch (getResponse.type) {
+          case StoreValueType.STRING:
+            console.log('The value is a string:', getResponse.value);
+            break;
+          case StoreValueType.INTEGER:
+            console.log('The value is an integer:', getResponse.value);
+            break;
+          case StoreValueType.BOOLEAN:
+            console.log('The value is a boolean:', getResponse.value);
+            break;
+        }
+        console.log(`The value is: ${getResponse.value} and the type is ${getResponse.type}`);
+        break;
+      case StoreGet.ResponseType.Error:
+        console.error('Error:', getResponse.message);
+        break;
```

Comment:
can you add some comments in here? I want to know things like:

* is there a flag that you find that makes the compiler check for exhaustive branches?
* if you have that flag enabled and you omit a branch what happens?
* it seems that the trick here (and from that SO post that I linked on MIchael's) is to have a nested field that is basically an enum type that differentiates the different response types, so you'd always be switching on that field but then accessing other (sibling) fields of the response object in the case branches, does that match up with your understanding?
* If we decided we loved this but that it feels too inconsistent with the existing SDK, can you imagine a way we could augment the existing CacheResponse types to support this pattern? (I think "yes" but would like to see/hear your thoughts)

Reply:
1. Yes there is a rule here https://typescript-eslint.io/rules/switch-exhaustiveness-check/
2. This is what it looks like
![image](https://github.com/momentohq/client-sdk-javascript/assets/11823378/8947110b-af79-4a00-8b58-53e9f96bef6e)
3. Yes this is also my understanding, but the way that they do it in the stack overflow post, I am not sure if there is any way to guarantee exhaustive check like there is with the enum
4. I need to play around with this a bit before I can say for sure, I can do that and get back to you on this thread with an example if I can get it working